.github/workflows/ci.yml

```diff
@@ -1,32 +1,98 @@
 name: CI
 
-env:
-  COMPOSE_INTERACTIVE_NO_CLI: 1
-
 on:
   push:
-    branches: [ 3.x, 2.x, master ]
   pull_request:
     branches: [ 3.x, 2.x, master ]
 
 jobs:
   tests:
     runs-on: ubuntu-latest
+    container: shivammathur/node:latest
 
     strategy:
       matrix:
-        php: ["7.4", "8.0"]
-        laravel: ["^6.0", "^8.0"]
+        php: ['7.4', '8.0', '8.1']
+        laravel: ['^6.0', '^8.0']
+        exclude:
+          - php: '8.1'
+            laravel: '^6.0'
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: root
+          POSTGRES_DB: main
+        ports:
+          - 5432/tcp
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 3
+
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: main
+        ports:
+          - 3306/tcp
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+      mysql2:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: main
+        ports:
+          - 3306/tcp
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+      redis:
+        image: redis:alpine
+        ports:
+          - 6379/tcp
+        options: --health-cmd="redis-cli ping" --health-interval=10s --health-timeout=5s --health-retries=3
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Start docker containers
-      run: PHP_VERSION=${{ matrix.php }} docker-compose up -d
-    - name: Install dependencies
-      run: docker-compose exec -T test composer require --no-interaction "laravel/framework:${{ matrix.laravel }}"
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php }}
+        extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, redis, bcmath, soap, intl, gd, exif, iconv, imagick, fileinfo
+        coverage: pcov
+
+    - name: Get composer cache directory
+      id: composer-cache
+      run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+    - name: Cache composer dependencies
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+        restore-keys: ${{ runner.os }}-composer-
```

Comment:
What's this step for?

Reply:
Is the cache PHP/Laravel version-specific?

Reply:
@abrardev99 I think you addressed this, but the review is still open. Is this resolved now? If so - how?

Reply:
Putting the [below comments conversation](https://github.com/archtechx/tenancy/pull/793#issuecomment-1197044280) here. 

>me: I think the cache key is unique based on `composer.json. key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}`.

>you: This would mean that it's not unique on each PHP/Laravel version.

I think it is unique based on the `composer.lock` file because each Laravel installation would change the `composer.lock` so a new cache is automatically created. 
I'm getting this point from the [docs](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#using-contexts-to-create-cache-keys).
<img width="758" alt="image" src="https://user-images.githubusercontent.com/54532330/181452030-da4dccd6-8748-4e89-9ebb-394333980f01.png">

If you still think this is not working correctly, we can use the `matrxi.key` combination.